lang/LangPrimSource/PyrMathPrim.cpp

```diff
@@ -862,7 +862,7 @@ int prSimpleNumberSeries(struct VMGlobals* g, int numArgsPushed) {
         }
 
         step = second - first;
-        size = (int)floor((last - first) / step + 0.001) + 1;
```

Comment:
You're right to catch that this line creates a bug, but I also worry that it prevents a different bug. Is it now possible to create a divide-by-zero error here with a zero value for step? I tested this patch with a step=0 and I know the error check on line 867 is catching this, but would it make sense to add a special check for a zero step size and report that error before attempting the divide?

Reply:
Hey @mtmccrea thanks for the PR! I had one question but overall looks good.

Reply:
Thanks for having a look at this @lnihlen!

Good catch. I tested the 0-step case and it was caught as you saw, but it's a good point about the divide by zero.

Out of curiosity, I inserted some checks to see what the math was doing:
```
        step = second - first;
		post(
			 "prSimpleNumberSeries: first: %f, step: %f, last %f\n",
			 first, step, last);
        size = (int)((last - first) / step) + 1;
		post(
			 "prSimpleNumberSeries: size: %d\n", size);
		
        if ((size < 1) || ((step >= 0) && (last < first)) || ((step <= 0) && (last > first))) {
            post(
                "prSimpleNumberSeries: arguments do not form an arithmetic progression: first: %f, step: %f, last %f\n",
                first, step, last);
            return errFailed;
        }
```
and with this input:
`(1.2, 1.2 .. 100)`
here's what went through:
```
prSimpleNumberSeries: first: 1.200000, step: 0.000000, last 100.000000
prSimpleNumberSeries: size: -2147483647
prSimpleNumberSeries: arguments do not form an arithmetic progression: 
first: 1.200000, step: 0.000000, last 100.000000
```

So yes, worth catching.
I found that there was an inconsistency between `Int` and `Float` behavior:
```
/* 0-step, first == last */
1.series(1, 1)
// -> [ 1 ]
1.2.series(1.2, 1.2)
// -> ERROR: Primitive '_SimpleNumberSeries' failed.
```
So rather than error out on a step of `0`, I now catch the case and set `size` to `1`. So the behavior is now consistent with `Int`:
```
/* 0-step, first == last */
1.series(1, 1)
// -> [ 1 ]
1.2.series(1.2, 1.2)
// -> [ 1.2 ]
```

I'll push this change shortly.
Thanks!

Reply:
I've also added a unit test for this case.